syntax = "proto2";

package watrix.proto;

message CameraImage {
  enum ImageType {
    ORIGIN = 0;
    MASK_RESULT = 1;
    YOLO_DETECT_RESULT = 2;
  }
  
  optional uint32 camera_id = 1; 
  optional uint32 height    = 2;  
  optional uint32 width     = 3;  
  optional uint32 channel   = 4;  
  optional ImageType type   = 5;
  optional bytes  data      = 6;
  optional uint64 timestamp_msec = 7;  
}

message DetectionBox {
  optional uint32 xmin = 1;
  optional uint32 ymin = 2;
  optional uint32 xmax = 3;
  optional uint32 ymax = 4;
  optional float confidence = 5;
  optional uint32 class_index = 6;
  optional bytes class_name = 7;
}

// for 1 image, we get 1 DetectionBoxs (N DetectionBox)
message DetectionBoxs {
  repeated DetectionBox boxs = 1; 
}

message YoloDetectionResult {
  optional CameraImage source_image = 1;
  optional DetectionBoxs detection_boxs = 2;
}

message TrainSegResult {
  optional CameraImage source_image = 1; 
  optional CameraImage binary_mask = 2; 
}

message SyncCameraResult {
  optional CameraImage sync_image = 1; 
  optional uint64 timestamp_msec = 2;  
}